adminapi/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -613,6 +619,257 @@ public Message getServerInfo() throws IOException, NoSuchAlgorithmException, Inv
+    if (description != null && !description.isEmpty()) {
+      serviceAccount.put("description", description);
+    }
+    if (comment != null && !comment.isEmpty()) {
+      serviceAccount.put("comment", comment);
+    }
+    String expirationFormat = null;
+    if (expiration != null) {
+      expirationFormat = expiration.format(Time.EXPIRATION_DATE_FORMAT);
+      serviceAccount.put("expiration", expirationFormat);
+    }
+    Credentials creds = getCredentials();
+    try (Response response =
+        execute(
+            Method.PUT,
+            Command.ADD_SERVICE_ACCOUNT,
+            ImmutableMultimap.of("accessKey", accessKey),
+            Crypto.encrypt(creds.secretKey(), OBJECT_MAPPER.writeValueAsBytes(serviceAccount)))) {
+      return new Credentials(
+          accessKey,
+          secretKey,
+          null,
+          (expiration != null) ? ResponseDate.fromString(expirationFormat) : null);
+    }
+  }
+
+  /**
+   * Edit an existing service account.
+   *
+   * <pre>Example:{@code
+   * // Assume policyJson contains below JSON string;
+   * // {
+   * //     "Statement": [
+   * //         {
+   * //             "Action": "s3:GetObject",
+   * //             "Effect": "Allow",
+   * //             "Principal": "*",
+   * //             "Resource": "arn:aws:s3:::my-bucketname/myobject*"
+   * //         }
+   * //     ],
+   * //     "Version": "2012-10-17"
+   * // }
+   * //
+   * }</pre>
+   *
+   * @param accessKey Access key.
+   * @param newStatus New service account status.
+   * @param newSecretKey New secret key.
+   * @param newPolicy New policy as JSON string .
+   * @param newName New service account name.
+   * @param newDescription New description.
+   * @param newExpiration New expiry time , Example : 2023-12-02T15:04:05Z.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws IOException thrown to indicate I/O error on MinIO REST operation.
+   * @throws InvalidCipherTextException thrown to indicate data cannot be encrypted/decrypted.
+   */
+  public void updateServiceAccount(
+      @Nullable String newName,
+      @Nullable String newSecretKey,
+      @Nonnull String accessKey,
+      @Nullable boolean newStatus,
+      @Nullable String newPolicy,
+      @Nullable String newDescription,
+      @Nullable ZonedDateTime newExpiration)
+      throws NoSuchAlgorithmException, InvalidKeyException, IOException,
+          InvalidCipherTextException {
+    if (newDescription != null && newDescription.length() > 256) {
+      throw new IllegalArgumentException("description must be at most 256 characters long");
+    }
+    if (accessKey == null || accessKey.isEmpty()) {
+      throw new IllegalArgumentException("access key must be provided");
+    }
+    Map<String, Object> serviceAccount = new HashMap<>(7);
+    serviceAccount.put("newStatus", newStatus ? "on" : "off");
```

Comment:
I do not see `on`/`off` fixed value in `madmin-go`

Reply:
The source of this value is the implementation of the referenced mc client.
https://github.com/minio/mc/blob/master/cmd/admin-user-svcacct-disable.go#L72
I suggest keeping consistent with the mc client.